products/cloudbuild/api.yaml

```diff
@@ -503,3 +503,186 @@ objects:
                     have completed successfully. If `wait_for` is empty, this build step
                     will start when all previous build steps in the `Build.Steps` list
                     have completed successfully.
+          - !ruby/object:Api::Type::NestedObject
+            name: 'artifacts'
+            description: |
+              Artifacts produced by the build that should be uploaded upon successful completion of all build steps.
+            properties:
+              - !ruby/object:Api::Type::Array
+                name: 'images'
+                item_type: Api::Type::String
+                description: |
+                  A list of images to be pushed upon the successful completion of all build steps.
+
+                  The images will be pushed using the builder service account's credentials.
+
+                  The digests of the pushed images will be stored in the Build resource's results field.
+
+                  If any of the images fail to be pushed, the build is marked FAILURE.
+              - !ruby/object:Api::Type::NestedObject
+                name: 'objects'
+                description: |
+                  A list of objects to be uploaded to Cloud Storage upon successful completion of all build steps.
+
+                  Files in the workspace matching specified paths globs will be uploaded to the
+                  Cloud Storage location using the builder service account's credentials.
+
+                  The location and generation of the uploaded objects will be stored in the Build resource's results field.
+
+                  If any objects fail to be pushed, the build is marked FAILURE.
+                properties:
+                  - !ruby/object:Api::Type::String
+                    name: 'location'
+                    description: |
+                      Cloud Storage bucket and optional object path, in the form "gs://bucket/path/to/somewhere/".
+
+                      Files in the workspace matching any path pattern will be uploaded to Cloud Storage with
+                      this location as a prefix.
+                  - !ruby/object:Api::Type::Array
+                    name: 'paths'
+                    item_type: Api::Type::String
+                    description: |
+                      Path globs used to match files in the build's workspace.
+                  - !ruby/object:Api::Type::NestedObject
+                    name: 'timing'
+                    description: |
+                      Output only. Stores timing information for pushing all artifact objects.
+                    output: true
+                    properties:
+                      - !ruby/object:Api::Type::String
+                        name: 'startTime'
```

Comment:
This and endTime also need to be marked output: true. You can see at https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-3944-old..auto-pr-3944#diff-f0c51a3d504e195b74aeb122b2747c9eR302-R317 that it isn't automatic.

Reply:
I'm not sure what you mean by that. I just tried setting `output: true` on them locally and it did exactly what I expected:
```
                  - !ruby/object:Api::Type::NestedObject
                    name: 'timing'
                    description: |
                      Output only. Stores timing information for pushing all artifact objects.
                    output: true
                    properties:
                      - !ruby/object:Api::Type::String
                        name: 'startTime'
                        output: true
                        description: |
                          Start of time span.

                          A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to
                          nine fractional digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
                      - !ruby/object:Api::Type::String
                        name: 'endTime'
                        output: true
                        description: |
                          End of time span.

                          A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to
                          nine fractional digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
```